File: jwt/src/main/java/com/member/jwt/service/CustomMemberDetailService.java
package com.member.jwt.service;

import com.member.jwt.dto.CustomUserDetails;
import com.member.jwt.entity.MemberEntity;
import com.member.jwt.repository.MemberRepository;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
public class CustomMemberDetailService implements UserDetailsService {

    private final MemberRepository memberRepository;

    public CustomMemberDetailService(MemberRepository memberRepository) {

        this.memberRepository = memberRepository;
    }

    @Override
    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {

        MemberEntity memberEntity = memberRepository.findByEmail(email)
                .orElseThrow(() -> new UsernameNotFoundException("사용자를 찾을 수 없습니다."));

        return new CustomUserDetails(memberEntity);
    }
}

File: jwtKotlin/src/main/kotlin/com/member/jwtkotlin/service/CustomMemberDetailService.kt
package com.member.jwtkotlin.service

import com.member.jwtkotlin.dto.CustomUserDetails
import com.member.jwtkotlin.repository.MemberRepository
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service
import java.util.function.Supplier

@Service
class CustomMemberDetailService(private val memberRepository: MemberRepository) : UserDetailsService {
    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(email: String): UserDetails {
        val memberEntity = memberRepository.findByEmail(email)
            ?.orElseThrow(Supplier { UsernameNotFoundException("사용자를 찾을 수 없습니다.") })!!

        return CustomUserDetails(memberEntity)
    }
}